task/backend/scheduler_metrics.go

```diff
@@ -71,12 +80,15 @@ func (sm *schedulerMetrics) PrometheusCollectors() []prometheus.Collector {
 		sm.runsActive,
 		sm.claimsComplete,
 		sm.claimsActive,
+		sm.queueDelta,
 	}
 }
 
 // StartRun adjusts the metrics to indicate a run is in progress for the given task ID.
-func (sm *schedulerMetrics) StartRun(tid string) {
+// We are also storing the delta time between when a run is due to start and actually starting.
+func (sm *schedulerMetrics) StartRun(tid string, queueDelta float64) {
```

Comment:
Rather than leak prometheus' types into this StartRun function would you take a time.Duration instead?  Then you could do the conversion as close to the prom library as possible.